build-tools/automation/yaml-templates/apk-instrumentation.yaml

```diff
@@ -0,0 +1,29 @@
+parameters:
+  configuration: []
+  testName: []
+  project: []
+  testResultsFiles: []
+  extraBuildArgs: ""
+  testResultsFormat: NUnit
+  packageType: Apk
+
+steps:
+- task: MSBuild@1
+  displayName: run ${{ parameters.testName }}
+  inputs:
+    solution: ${{ parameters.project }}
+    configuration: ${{ parameters.configuration }}
+    msbuildArguments: >
+      /t:AcquireAndroidTarget,SignAndroidPackage,UndeployTestApks,DeployTest${{ parameters.packageType }}s,RunTestApks,RenameApkTestCases,ReportComponentFailures
```

Comment:
Ouch; this means *each `apk-instrumentation.yaml` invocation* will ***separately*** create and launch the Android emulator, which is not "fast", and is done *11* times.

I shudder to think at how much CPU time we'll lose from recreating and relaunch the emulator 11 times, never mind actually running any unit tests...

Would it be possible to instead do a single `/t:AcquireAndroidTarget` *before* performing any installs, then call `/t:ReleaseAndroidTarget` *after* all the on-device tests have executed?

Reply:
Since we're not releasing the target until the end the emulator will stay alive, the performance on the hosted mac pool was actually pretty good before I broke things after rebasing.